R/messages.R

```diff
@@ -4,58 +4,176 @@
+  # Confirm that the server address ends in a /
+  if (!is.na(server)) {
+    last_char <- substr(server, (nchar(server)+1)-1, nchar(server))
+    if (!is.na(last_char) && last_char != "/") {
+      server <- paste0(server, "/")
     }
+  }
 
+  url <- paste0(server, "__api__/server_settings")
+  server_settings <- NULL
+  result <- tryCatch (
+    {
+      "!DEBUG Sending GET request to `url`"
+      response <- httr::GET(
+        url,
+        httr::add_headers(Authorization = paste0("Key ", api_key)),
+        httr::write_memory()
+      )
+      list(success=TRUE, response=response)
+    },
+    error = function(err) {
+      "!DEBUG GET response threw an exception: `err`"
+
+      return (list(
+        success=FALSE, 
+        message=paste0("### API request to ", server, " has failed with error:",
+            "\n", err, 
+            "\n",
+            "\nPossible Solutions:",
+            "\n",
+            "\n-    If you have specified an API_KEY, confirm it is valid.",
+            "\n-    Confirm there is connectivity between the server itself and the address assigned to it: ", server, ".", 
+            "\n-    If using HTTPS along with self-signed certificates, you may need to allow the plumbertableau package to use HTTP instead, ", 
+            "by setting the environment variable *PLUMBERTABLEAU_USE_HTTP* to *TRUE* within the RStudio Connect application settings."
+        )
+      ))
+    }
+  )
+  if (!result$success) {
+    "!DEBUG Detected that an error has been returned from exception: `result`"
 
-    # Send warning message to the console if any of the above are TRUE
-    if (!rlang::is_null(message_contents)) {
-      message_contents <- paste(
+    message_contents <- paste0(
+      message_contents,
+      result$message,
+      sep = "\n\n---\n\n"
+    )
+    message_count <- message_count + 1
+  } else {
+    "!DEBUG GET response has returned `httr::http_status(result$response)$category`, `httr::http_status(result$response)$reason`, `httr::http_status(result$response)$message`"
+    
+    if (httr::http_error(result$response)) {
+      message_contents <- paste0(
         message_contents,
-        "\n\n#### Please reach out to your RStudio Connect administrator to fix these issues.",
-        sep = "\n"
+        paste0("### API request to ", server, " failed. ",
+        "\n-    Response: ", httr::http_status(result$response)$reason, ", ", httr::http_status(result$response)$message,
+        "\n",
+        "\nPossible Solution:",
+        "\n",
+        "\nDiagnose connectivity or access issue."
+        ),
+        sep = "\n\n---\n\n"
       )
+      message_count <- message_count + 1
 
-      rlang::warn(stringi::stri_replace_all(message_contents, regex = "\\* ?|#+ ", replacement = ""), .frequency = "once", .frequency_id = "rsc_warning")
+    } else {
+      server_settings <- httr::content(result$response, as="parsed")
+      "!DEBUG GET response was successful. Server settings = `server_settings`"
```

Comment:
I'm a bit hesitant about logging all RSC server settings, since there may be details in there an RSC admin doesn't want individual users to know.

Reply:
It's also worth noting that `DEBUGME` logs the resulting vector by creating many copies of the full log message:
```
2021/07/28 15:50:20.064376600 plumbertableau +-GET response was successful. Server settings = dev-saml +5ms plumbertableau +-GET response was successful. Server settings = 1.8.8.3-dev665-dirty +5ms plumbertableau +-GET response was successful. Server settings = v1.8.8.2-665-gd2a9e58-dirty +5ms plumbertableau +-GET response was successful. Server settings = RStudio Connect v1.8.8.3-dev665-dirty +5ms plumbertableau +-GET response was successful. Server settings = list(handles_credentials = FALSE, handles_login = TRUE, challenge_response_enabled = FALSE, external_user_data = TRUE, external_user_search = FALSE, external_user_id = TRUE, groups_enabled = TRUE, external_group_search = FALSE, external_group_members = TRUE, external_group_id = FALSE, external_group_owner = TRUE, unique_usernames = FALSE, name_editable_by = "provider", email_editable_by = "provider", username_editable_by = "provider", role_editable_by = "adminandself", name = "SAML", notice = "", 
2021/07/28 15:50:20.064388700     warning_delay = 10) +5ms plumbertableau +-GET response was successful. Server settings = list(ts = 1627440035333, status = "activated", expiration = 1.630368e+12, `days-left` = 34, edition = "", cores = "0", connections = "0", `has-key` = TRUE, `has-trial` = FALSE, type = "local", `shiny-users` = "20", users = "20", `user-activity-days` = "365", `allow-apis` = "1") +5ms plumbertableau +-GET response was successful. Server settings =  +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings =  +5ms plumbertableau +-GET response was successful. Server settings =  +5ms plumbertableau +-GET response was successful. Server settings = __logout__ +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = list("connect", "apps", "users", "groups", "setpassword", "user-completion", "confirm", "recent", "reports", "plots", "unpublished", "settings", "metrics", "tokens", "help", "login", "welcome", "register", "resetpassword", "content") +5ms plumbertableau +-GET response was successful. Server settings = permissive +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = list("R", "Python") +5ms plumbertableau +-GET response was successful. Server settings = compact +5ms plumbertableau +-GET response was successful. Server settings = 10000000 +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = FALSE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = https://dev-saml.localtest.me/ +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms plumbertableau +-GET response was successful. Server settings = TRUE +5ms 
```

Reply:
Removed.